includes/class-boldgrid-backup-restorer.php

```diff
@@ -0,0 +1,243 @@
+<?php
+/**
+ * File: class-boldgrid-backup-restorer.php
+ *
+ * @link https://www.boldgrid.com
+ * @since SINCEVERSION
+ *
+ * @package    Boldgrid_Backup
+ * @subpackage Boldgrid_Backup/includes
+ * @copyright  BoldGrid
+ * @version    $Id$
+ * @author     BoldGrid <support@boldgrid.com>
+ */
+
+/**
+ * Class: Boldgrid_Backup_Restorer
+ *
+ * Originally, all methods for restoring a file have lived in Boldgrid_Backup_Admin_Core. This class,
+ * over time, will absorb those methods.
+ *
+ * @since SINCEVERSION
+ */
+class Boldgrid_Backup_Restorer {
+	/**
+	 * Admin core.
+	 *
+	 * @since SINCEVERSION
+	 * @access private
+	 * @var Boldgrid_Backup_Admin_Core
+	 */
+	private $core;
+
+	/**
+	 * An array of info about our archive.
+	 *
+	 * @since SINCEVERSION
+	 * @access private
+	 * @var array
+	 */
+	private $info;
+
+	/**
+	 * An instance of Boldgrid_Backup_Admin_Task.
+	 *
+	 * @since SINCEVERSION
+	 * @access private
+	 * @var Boldgrid_Backup_Admin_Task
+	 */
+	private $task;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public function __construct() {
+		$this->core = apply_filters( 'boldgrid_backup_get_core', null );
+	}
+
+	/**
+	 * Steps to take when archiving is complete.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public function complete() {
+		// Update the log.
+		$this->core->logger->add( 'Restoration complete!' );
+		$this->core->logger->add_memory();
+
+		if ( $this->has_error() ) {
+			$this->core->logger->add( 'Error during restoration: ' . $this->get_error() );
+		}
+
+		// End the task.
+		$this->task->end();
+	}
+
+	/**
+	 *
+	 */
+	private function get_error() {
+		return ! empty( $this->info['error'] ) ? $this->info['error'] : __( 'Unknown error', 'boldgrid-backup' );
+	}
+
+	/**
+	 * Get our archive info.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return array
+	 */
+	public function get_info() {
+		return $this->info;
+	}
+
+	/**
+	 *
```

Comment:
I feel like there should be something here, but I can't quite place my finger on it.

Reply:
https://github.com/BoldGrid/boldgrid-backup/commit/536cd31b31cf1cbd2c30293aae264e3f4b16147d